homeassistant/components/pooldose/config_flow.py

```diff
@@ -25,70 +26,95 @@
-                step_id="user",
-                data_schema=SCHEMA_DEVICE,
-                errors={"base": "api_not_supported"},
-                description_placeholders=api_versions,
-            )
+            return None, api_versions, {"base": "api_not_supported"}
 
         device_info = client.device_info
         if not device_info:
+            return None, None, {"base": "no_device_info"}
+        serial_number = device_info.get("SERIAL_NUMBER")
+        if not serial_number:
+            return None, None, {"base": "no_serial_number"}
+
+        return serial_number, None, None
+
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle a simple DHCP discovery: abort if already configured, else show a confirmation dialog."""
+
+        # test outputs, use MAC later to complete the connection attribute of existing devices
+        _LOGGER.debug("DHCP discovery IP address  %s", discovery_info.ip)
+        _LOGGER.debug("DHCP discovery MAC address %s", discovery_info.macaddress)
+        serial_number, _, _ = await self._validate_host(discovery_info.ip)
+        if not serial_number:
+            return self.async_abort(reason="no_serial_number")
+
+        await self.async_set_unique_id(serial_number)
+        self._abort_if_unique_id_configured()
+        self._discovered_ip = discovery_info.ip
+        return self.async_show_form(
+            step_id="dhcp_confirm",
+            description_placeholders={
+                "ip": discovery_info.ip,
+                "mac": discovery_info.macaddress,
+                "name": f"PoolDose {serial_number}",
+            },
+        )
+
+    async def async_step_dhcp_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Create the entry after the confirmation dialog."""
+        discovered_ip = self._discovered_ip
+        return self.async_create_entry(
+            title=f"PoolDose {self.unique_id}",
+            data={CONF_HOST: discovered_ip},
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        if not user_input:
             return self.async_show_form(
                 step_id="user",
                 data_schema=SCHEMA_DEVICE,
-                errors={"base": "no_device_info"},
             )
-        serial_number = device_info.get("SERIAL_NUMBER")
-        if not serial_number:
+
+        host = user_input[CONF_HOST]
+        serial_number, api_versions, errors = await self._validate_host(host)
+        if errors:
             return self.async_show_form(
                 step_id="user",
                 data_schema=SCHEMA_DEVICE,
-                errors={"base": "no_serial_number"},
+                errors=errors,
+                description_placeholders=api_versions,
             )
 
         await self.async_set_unique_id(serial_number)
```

Comment:
```suggestion
        await self.async_set_unique_id(serial_number, raise_on_progress=False)
```
Otherwise the user flow would raise if there is a dhcp flow already discovered